### DISCUSSION TITLE
Struggling to get Keycloak 22.0.3 to work with Springboot 3.1.5

### DISCUSSION THREAD

**Discussion by lordarnoud** [keycloak/keycloak#24822] [Q&A]
Hello!

I am not sure if I am allowed to ask this here but I will try anyway! 
Ever since the deprecation of adapters I haven't been able to find any good resource on how to implement **Keycloak 22.0.3** in my **Java 21**, **Springboot 3.1.5** application with **Vaadin 24.2.2**. I know the Vaadin part is not super relevant, but I say it to indicate it's a client for users and not some kind of backend thing. 

**What am I trying to do?**
I am trying to get a login page to show up and then later check based on the role of the logged in user if they have access to a page or not. My services, model and UI are all within this 1 applications. My application properties are defined in my **application.properties** file.

A lot of the tutorials I find are deprecated and don't even compile. Could anyone here please help me out on how to use Keycloak to get a login page on my application?

**The hard parts** for my final goal that might be totally out of place here but I feel like I should mention them as it is what I desire to make and I really do need some help with it:
- I would like to check the login of the users to my own **MySQL8** database as we have our own password encryption.
- I would like to check the role of the user based on the role in my database, but I wouldn't mind copying/linking these roles to Keycloak.
- I would like to be able to call some kind of method in the code to get the information of the logged in user, so not using annotations, but using a service/method.

I know this might not be the place to ask, or it might be too much, but I am shooting my shot! Maybe this is something simple that I am just not being able to find.

**Things I have tried:**
- https://stackoverflow.com/questions/74571191/use-keycloak-spring-adapter-with-spring-boot-3
- https://www.youtube.com/watch?v=_0oXZKr97ro&t=988s&ab_channel=ProgrammingTechie
- https://www.baeldung.com/spring-boot-keycloak
- and some others that were a bit too short/bad to mention.

My own StackOverflow question which is a bit different from this post as originally. 
https://stackoverflow.com/questions/77486271/how-to-use-keycloak-22-and-springboot-3-1-5-and-spring-security-6-1-5-all-togeth
I had a different question but they made me edit it because they believed that the first link  (https://stackoverflow.com/questions/74571191/use-keycloak-spring-adapter-with-spring-boot-3) solved all my answers so my question was considered a duplicate.

But with all of these the code didn't even compile and was already shown in my IDE to be outdated.

**Comment by mira13** [comment]
Hello, have you found a solution eventually?

**Reply by unclevlad** [reply]
Sadly, I am right there with you. The amount of outdated code for this is staggering.

**Reply by lordarnoud** [reply]
Oops sorry I totally missed this message! But no, I ended up giving up and just used the normal spring security :') I threw out anything Keycloak related :(

**Reply by ThoreKr** [reply]
What upsets you about this approach? After all, spring security is the recommended way to go https://www.keycloak.org/docs/latest/securing_apps/#_spring_boot_adapter and you are using an open standard (oidc) with a well estabilshed and well maintained security framework (spring security). To me, these are the things i'd be looking for when choosing an authentication solution?

**Reply by lordarnoud** [reply]
Well I still wanted to be able to use my Keycloak to authenticate the users and permissions with. And they do say it's going to be deprecated, but they don't really help out on telling us how to migrate away from it while still being able to use Keycloak to manage our users. At least me and my colleagues didn't manage to make it work after removing the adapter. There were still other Keycloak classes that were used in the code that suddenly didn't exist anymore either. Wouldn't know which ones as it's been too long since :( So maybe the guide is clearing now on how to migrate, but back then we didn't see or manage.

I am not against their choice of deprecating it, I just would've liked some more help on how to keep using Keycloak while integrating it in a Springboot application if you have been using their adapters :)
